fetch/cosmwasm_contract/src/contract.rs

```diff
@@ -233,24 +227,19 @@ fn try_reverse_swap(
             Ok(state)
         })?;
 
-        let attr = vec![
+        let attrs = vec![
             attr("action", "reverse_swap"),
-            attr("rid", rid),
+            attr("rid", rid.to_string()),
             attr("to", to),
             attr("sender", sender),
             attr("origin_tx_hash", origin_tx_hash),
             attr("amount", effective_amount),
             attr("swap_fee", swap_fee),
         ];
         // FIXME(LR) store revese swap id similarly to refunds?
-
-        let r = Response {
-            messages: rtx.messages,
-            submessages: vec![],
-            attributes: attr,
-            data: None,
-        };
-        Ok(r)
+        Ok(Response::new()
+            .add_attributes(attrs)
+            .add_submessages(rtx.messages))
```

Comment:
This is a functional difference right? Did you expect to call `.add_message` it might not make much of the difference but my recollection is that sub messages are ones where the contract is interested in the response.

Reply:
I think I'm forced to use `.add_submessages` here because of message signature: `pub messages: Vec<SubMsg<T>, Global>`. Using `.add_message` result in `the trait bound `cosmwasm_std::CosmosMsg<_>: From<SubMsg>` is not satisfied
`